lib/trino-parquet/src/main/java/io/trino/parquet/reader/PageReader.java

```diff
@@ -175,13 +187,19 @@ public void skipNextPage()
         compressedPages.next();
     }
 
-    public boolean arePagesCompressed()
+    private void verifyDictionaryPageRead()
     {
-        return codec != CompressionCodec.UNCOMPRESSED;
+        checkArgument(dictionaryAlreadyRead, "Dictionary has to be read first");
     }
 
-    private void verifyDictionaryPageRead()
+    private OutputBuffer createOutputBuffer()
     {
-        checkArgument(dictionaryAlreadyRead, "Dictionary has to be read first");
+        return size -> {
+            if (decompressorOutputBuffer == null || size > decompressorOutputBuffer.length) {
+                decompressorOutputBuffer = new byte[size];
```

Comment:
Can there be a rogue `size` that would dramatically increase reader memory usage? Should buffer be shrink after a while?

Reply:
This wouldn't have been problem as page sizes are stable within a column chunk (every writer is aiming for some size and values count threshold).